typescript/sdk/src/deploy/verify/utils.ts

```diff
@@ -64,6 +68,45 @@ export function getContractVerificationInput({
   );
 }
 
+export async function getContractVerificationInputForZKSync({
+  name,
+  contract,
+  constructorArgs,
+  artifact,
+  isProxy,
+  expectedimplementation,
+}: {
+  name: string;
+  contract: ethers.Contract;
+  constructorArgs: any[];
+  artifact: ZkSyncArtifact;
+  isProxy?: boolean;
+  expectedimplementation?: Address;
+}): Promise<ContractVerificationInput> {
+  const args = await encodeArguments(artifact.abi, constructorArgs);
+  return buildVerificationInput(
+    name,
+    contract.address,
+    args,
+    isProxy,
+    expectedimplementation,
+  );
+}
+
+export function encodeArguments(abi: any, constructorArgs: any[]): string {
+  const contractInterface = new Interface(abi);
+  let deployArgumentsEncoded;
+  try {
+    deployArgumentsEncoded = contractInterface
+      .encodeDeploy(constructorArgs)
+      .replace('0x', '');
```

Comment:
I wonder if this is related to the other thing I was running into

Reply:
Nahh, I doubt it, its used in `getContractVerificationInputForZKSync`. IIRC you had an issue with `quoteDispatch`